### PR TITLE
LINK-1648 | 2/3 Attendance list page action buttons

### PR DESCRIPTION
**Pull request by jorilindell**
## Description
Adds action menu dropdown to attendance list page. At the moment attendance menu dropdown contains only a button to open signups page. There will be also a button to download signups as an excel file

This one of the 3 PRs that will implement signups page. PRs are splitted to following parts:
- MenuDropdown component (https://github.com/City-of-Helsinki/linkedregistrations-ui/pull/95)
- **Attendance list page action buttons**
- Signups page (https://github.com/City-of-Helsinki/linkedregistrations-ui/pull/97)

## Partially closes
[LINK-1648](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1648)

## Screenshot
Screenshot 2023-10-26 at 20 30 58

[LINK-1648]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1648?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/domain/singups/signupsPagePermissions/SignupsPagePermissions.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    if (!session) {
+      return <SignInRequired />;
+    }
+
+    // Show strong identification required page if user doesn't have permissions to edit signups
+    if (!user?.is_strongly_identified) {
+      return <StrongIdentificationRequired />;
+    }
+
+    // Show not found page if user doesn't have permissions to edit signup oresence status
```
Comment (harriris-vincit) [inline src/domain/singups/signupsPagePermissions/SignupsPagePermissions.tsx:50]:
```suggestion
    // Show not found page if user doesn't have permissions to edit signup presence status
```

File: src/domain/registration/permissions.ts [hunk 2]
```diff
@@ -13,3 +14,24 @@ export const canUserUpdateSignupPresenceStatus = ({
...
+export const checkCanUserDoRegistrationAction = ({
+  action,
+  registration,
+  user,
+}: {
+  action: REGISTRATION_ACTIONS;
+  registration?: Registration;
+  user?: User;
+}): boolean => {
+  const isRegistrationUser = hasRegistrationUserAccess({
```
Comment (harriris-vincit) [inline src/domain/registration/permissions.ts:27]: Are superuser and registration admin permissions handled somewhere else or should they also be allowed to do the actions here?

Comment (jorilindell) [reply src/domain/registration/permissions.ts:27]: Superuser and registration admin users are using these same features from Linked Events UI e.g. https://linkedevents.dev.hel.ninja/fi/registrations/29/signups?returnPath=%2Fregistrations. Permissions for them are checked there. The reason to have separate page for registration user access user it that Linked Events UI doesn't support Suomi.fi authentication so users are not strongly identified.

File: src/domain/singups/signupsPagePermissions/SignupsPagePermissions.tsx (hunk 1 above)
Comment (jorilindell) [reply src/domain/singups/signupsPagePermissions/SignupsPagePermissions.tsx]: Fixed the typo

File: src/domain/singups/signupsPagePermissions/SignupsPagePermissions.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    // Show strong identification required page if user doesn't have permissions to edit signups
+    if (!user?.is_strongly_identified) {
+      return <StrongIdentificationRequired />;
+    }
+
+    // Show not found page if user doesn't have permissions to edit signup presence status
+    if (!hasRegistrationUserAccess({ registration, user })) {
+      return <NotFound />;
```
Comment (harriris-vincit) [inline src/domain/singups/signupsPagePermissions/SignupsPagePermissions.tsx:52]: A component about missing permissions would provide better feedback to the user (something similar to StrongIdentificationRequired)

Comment (jorilindell) [reply src/domain/singups/signupsPagePermissions/SignupsPagePermissions.tsx]: Added a new `InsufficientPermissions` component to display more detailed error page in the case that permissions are missing.

**Review by harriris-vincit — APPROVED** [review]
Looks really good to me

**Pull request merged**